### PR TITLE
infra: change travis arch to not use 'paid' credits

### PR DESCRIPTION
**Pull request by romani**
in context https://github.com/checkstyle/checkstyle/issues/10778

not all problems are resolved, but some jobs are moved to new arch.

### CODE REVIEW

**Comment by romani** [comment]
new problem:
https://app.travis-ci.com/github/checkstyle/contribution/jobs/545935061#L4245
```
Testing Checkstyle started
Caught: java.lang.NoClassDefFoundError: org/apache/tools/ant/BuildLogger
java.lang.NoClassDefFoundError: org/apache/tools/ant/BuildLogger
  at diff.deleteDir(diff.groovy:662)
  at diff$_generateCheckstyleReport_closure4.doCall(diff.groovy:284)
  at diff.generateCheckstyleReport(diff.groovy:264)
  at diff$generateCheckstyleReport$10.callCurrent(Unknown Source)
  at diff.launchCheckstyleReport(diff.groovy:224)
  at diff$launchCheckstyleReport$6.callCurrent(Unknown Source)
  at diff.run(diff.groovy:30)
Caused by: java.lang.ClassNotFoundException: org.apache.tools.ant.BuildLogger
  ... 7 more
```

File: .travis.yml [hunk 1]
```diff
@@ -53,10 +64,24 @@ addons:
...
+          export PATH=$JAVA_HOME/bin:$PATH
+      fi
     - groovy --version
 
 script:
-  - xmlstarlet ed --inplace -d "//mirrors" /home/travis/.m2/settings.xml
+  - |
+    set -e
+    MVN_SETTINGS=${TRAVIS_HOME}/.m2/settings.xml
+      if [[ -f ${MVN_SETTINGS} ]]; then
```
Comment (rnveach) [inline .travis.yml:78]: Technically this shouldn't be indented since the previous line isn't an if or such.

File: .travis.yml [hunk 2]
```diff
@@ -25,16 +30,22 @@ matrix:
         - CMD="./.ci/travis.sh patch-diff-report-tool"
 
     - jdk: openjdk8
+      arch: amd64
+      dist: xenial
```
Comment (rnveach) [inline .travis.yml:34]: Just curious why we don't use this arch for everything since it seems to fix some issues? Is this not one of the free ones?

**Review by rnveach — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (romani) [reply .travis.yml:34]: No it is not free tier arch, that arch need paid credits to be used, we always need to ask for new credit. In next PR there is a fix

File: .travis.yml (hunk 1 above)
Comment (romani) [reply .travis.yml]: fixed

**Pull request merged**